ddl/ddl_api.go

```diff
@@ -3422,32 +3427,17 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
 		Columns: meta.Partition.Columns,
 		Enable:  meta.Partition.Enable,
 	}
+
 	genIDs, err := d.genGlobalIDs(len(spec.PartDefinitions))
 	if err != nil {
 		return nil, err
 	}
-	buf := new(bytes.Buffer)
 	for ith, def := range spec.PartDefinitions {
 		if err := def.Clause.Validate(part.Type, len(part.Columns)); err != nil {
 			return nil, errors.Trace(err)
 		}
 		// For RANGE partition only VALUES LESS THAN should be possible.
 		clause := def.Clause.(*ast.PartitionDefinitionClauseLessThan)
-		for _, expr := range clause.Exprs {
```

Comment:
Why just remove it? Could we refine the checking?

Reply:
As you see, there are no test cases break and I'm adding new test cases, so it's fine to remove it.